src/docfx/lib/ProcessUtility.cs

```diff
@@ -275,12 +282,15 @@ public static async Task<string> ReadFile(string path)
             return result;
         }
 
+        public static Task<T> ReadFile<T>(string path)
+            => ReadFile(path, stream => JsonUtility.Deserialize<T>(new StreamReader(stream).ReadToEnd()));
+
         public static async Task<T> ReadFile<T>(string path, Func<Stream, T> read)
         {
             T result = default;
             await RunInsideMutex(path, () =>
             {
-                using (var fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, 1024, FileOptions.SequentialScan))
+                using (var fs = new FileStream(path, FileMode.OpenOrCreate, FileAccess.Read, FileShare.Read, 1024, FileOptions.SequentialScan))
```

Comment:
This change the semantic of `ReadFile`, it should not attempt to create a file.

Reply:
will revert that